otx/algorithms/detection/tasks/task.py

```diff
@@ -0,0 +1,608 @@
+        return result
+
+    def __reduce__(self):
+        """Function reduce in OnHookInitialized."""
+        return (self.__class__, (id(self.task_instance),))
+
+
+class OTXDetectionTask:
+    """Task class for OTX detection."""
+
+    # pylint: disable=too-many-instance-attributes, too-many-locals
+    def __init__(self, task_environment: TaskEnvironment, output_path: Optional[str] = None):
+        self._task_config = DetectionConfig
+        self._task_environment = task_environment
+        self._task_type = task_environment.model_template.task_type
+        self._labels = task_environment.get_labels(include_empty=False)
+        self._hyperparams = task_environment.get_hyper_parameters(self._task_config)  # type: ConfigDict
+        self._train_type = self._hyperparams.algo_backend.train_type
+        self._model_dir = os.path.join(
+            os.path.abspath(os.path.dirname(self._task_environment.model_template.model_template_path)),
+            TRAIN_TYPE_DIR_PATH[self._train_type.name],
+        )
+        self._work_dir_is_temp = False
+        self._output_path = output_path
+        if self._output_path is None:
+            self._output_path = tempfile.mkdtemp(prefix="OTX-task-")
+            self._work_dir_is_temp = True
+        else:
+            os.makedirs(self._output_path, exist_ok=True)
+        self._anchors: Dict[str, int] = {}
+        self._time_monitor = None
+        self.on_hook_initialized = OnHookInitialized(self)
+        self._learning_curves = UncopiableDefaultDict(OTXLoggerHook.Curve)
+        self._model_label_schema = []  # type: List[LabelEntity]
+        self._resume = False
+        self._should_stop = False
+        self.cancel_interface = None
+        self.reserved_cancel = False
+        self._model_ckpt = None
+        self._precision = [ModelPrecision.FP32]
+        self._optimization_methods = []  # type: List[OptimizationMethod]
+        self._is_training = False
+
+        if hasattr(self._hyperparams, "postprocessing") and hasattr(
+            self._hyperparams.postprocessing, "confidence_threshold"
+        ):
+            self.confidence_threshold = self._hyperparams.postprocessing.confidence_threshold
+        else:
+            self.confidence_threshold = 0.0
+
+        # This should be removed
+        self.override_configs = {}  # type: Dict[str, str]
+
+        if task_environment.model is not None:
+            self._load_model()
+
+        if self._hyperparams.tiling_parameters.enable_tiling:
+            self.data_pipeline_path = os.path.join(self._model_dir, "tile_pipeline.py")
+        else:
+            self.data_pipeline_path = os.path.join(self._model_dir, "data_pipeline.py")
+
+        # This is for hpo, and this should be removed
+        self.project_path = self._output_path
+
+    def _load_model(self):
+        """Loading model from checkpoint."""
+
+        def _load_model_ckpt(model: Optional[ModelEntity]):
```

Comment:
As far as I understand, those inner functions in `_load_model()` are used in other task. Then, how about moving those function definition outside to make them usable by other tasks?

Reply:
Great point, I wrote in the explanation material that current OTXDetectionTask is summation of OTXBaseTask and OTXDetectionTask. Therefore this OTXDetectionTask should be splitted. Since this is preview, I merge them for convinence